File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        double max=Integer.MIN_VALUE;
        int windowSum=0;
        int windowStart=0;
        for (int windowEnd = 0; windowEnd < n; windowEnd++) {
            windowSum+=nums[windowEnd];
            if(windowEnd>=k-1){
                max=Math.max(max,windowSum);
                windowSum-=nums[windowStart];
                windowStart++;
            }
        }
        return (max==Integer.MIN_VALUE)?0:max/k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        L,R,N=0,k,len(nums)
          
        ff=sum(nums[0:k])/k
        tem=ff
        while R<N:
            ff=(ff-(nums[L])/k)
            ff=(ff+(nums[R])/k)
            L+=1
            R+=1
            tem=max(ff,tem)
         
        return tem